index.js

```diff
@@ -218,10 +219,20 @@ module.exports = {
           if (doc.archived && (mode === 'published') && (doc.parkedId !== 'trash')) {
             return false;
           }
+
+          // We don't want the published doc if its draft version is archived
+          if (mode === 'published' && self.archivedDrafts.has(doc.workflowGuid)) {
```

Comment:
There can be more than one locale so this needs to be specific to the locale's own draft/published pair.